### PR TITLE
Use keycloak token for oAuth requests

### PR DESCRIPTION
**Pull request by vinokurig**
Signed-off-by: Igor Vinokur <ivinokur@redhat.com>

### What does this PR do?
* Use dedicated authentication window for oAuth authentication request.
* Use keycloak token for oAuth `getToken()` and `getProviders()` resquests.

### What issues does this PR fix or reference?
see: https://github.com/eclipse/che/pull/16488
depends on https://github.com/eclipse/che-workspace-client/pull/30

### CODE REVIEW

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts [hunk 1]
```diff
@@ -151,6 +144,11 @@ export async function start() {
     });
     theia.commands.registerCommand(VIEW, () => {
         viewPublicKey(sshKeyManager);
+        che.oAuth.getProviders().then(providers => {
+            providers.forEach(provider => {
+                console.log(provider);
```
Comment (azatsarynnyy) [inline plugins/ssh-plugin/src/ssh-plugin-backend.ts:149]: Is it some leftover or it's really needed?

File: extensions/eclipse-che-theia-plugin-ext/src/browser/oauth-utils.ts [hunk 2]
```diff
@@ -30,56 +34,59 @@ export class OauthUtils {
...
+                eventEmitter.fire(undefined);
+            } else if (data.data.startsWith('status:') && this.oAuthPopup) {
+                this.oAuthPopup.postMessage('token:' + (this.machineToken ? this.machineToken : ''), '*');
+            }
+        });
+    }
+
+    async getUserToken(): Promise<string | undefined> {
+        if (this.userToken) {
+            return new Promise(async resolve => resolve(this.userToken));
```
[the remark this answers is not in the document]
Comment (vinokurig) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/oauth-utils.ts]: done

File: extensions/eclipse-che-theia-plugin-ext/src/browser/oauth-utils.ts [hunk 3]
```diff
@@ -7,19 +7,23 @@
...
     private apiUrl: string;
     private machineToken: string | undefined;
-    private readonly cheApiService: CheApiService;
-
-    constructor(container: interfaces.Container) {
-        this.envVariableServer = container.get(EnvVariablesServer);
-        this.cheApiService = container.get(CheApiService);
+    private oAuthPopup: Window | undefined;
+    private userToken: string | undefined;
+    private readonly event: Event<void>;
```
[the remark this answers is not in the document]
Comment (vinokurig) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/oauth-utils.ts]: done

File: extensions/eclipse-che-theia-plugin-ext/src/browser/oauth-utils.ts [hunk 4]
```diff
@@ -7,19 +7,23 @@
...
+    constructor(@inject(EnvVariablesServer) private readonly envVariableServer: EnvVariablesServer,
+        @inject(CheApiService) private readonly cheApiService: CheApiService) {
+        const eventEmitter = new Emitter<void>();
```
[the remark this answers is not in the document]
Comment (vinokurig) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/oauth-utils.ts]: donw

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts (hunk 1 above)
Comment (vinokurig) [reply plugins/ssh-plugin/src/ssh-plugin-backend.ts]: Removed the leftovers :)

**Comment by azatsarynnyy** [comment]
crw-ci-test

**Comment by azatsarynnyy** [comment]
crw-ci-test

**Comment by azatsarynnyy** [comment]
crw-ci-test

**Comment by vinokurig** [comment]
crw-ci-test

**Comment by azatsarynnyy** [comment]
crw-ci-test

**Comment by vinokurig** [comment]
crw-ci-test

**Comment by vinokurig** [comment]
crw-ci-test

**Comment by dmytro-ndp** [comment]
crw-ci-test

**Comment by dmytro-ndp** [comment]
crw-ci-test

**Pull request merged**